File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Literals/Literal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UvA.SoftCon.Questionnaire.AST.Literals
{
    public abstract class Literal<T>
    {
        public T Value
        {
            get;
            private set;
        }

        public Literal(T value)
        {
            Value = value;
        }
    }
}

File: Nian_Karl/SoftwareConstruction/src/org/uva/ql/ast/expression/literal/Literal.java
package org.uva.ql.ast.expression.literal;

import org.uva.ql.ast.CodePosition;
import org.uva.ql.ast.expression.Expression;
import org.uva.ql.ast.value.Value;

public abstract class Literal extends Expression{
	
	public Literal(CodePosition pos) {
		super(pos);
	}
	
	public abstract <T extends Value> T getValue();

}
